src/librustc/hir/map/mod.rs

```diff
@@ -79,6 +79,33 @@ impl<'hir> Entry<'hir> {
         }
     }
 
+    fn fn_header(&self) -> Option<&'hir FnHeader> {
+        match &self.node {
+            Node::Item(item) => {
+                match &item.kind {
+                    ItemKind::Fn(_, fn_header, _, _) => Some(fn_header),
```

Comment:
Aside: I'm looking at [`hir::ItemKind::Fn`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/enum.ItemKind.html#variant.Fn) and [`hir::TraitItemKind::Method`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc/hir/enum.TraitItemKind.html#variant.Method) and it seems to me that `hir::ItemKind::Fn` could use `hir::MethodSig` as well (probably renamed to `hir::FnSig` to be more descriptive) to make the logic more regular here.

I was thinking of filing a C-cleanup issue for that... thoughts? (cc also @eddyb & @oli-obk)

Reply:
(I'm fixing this in https://github.com/rust-lang/rust/pull/66188.)